tests/test_transpose.py

```diff
@@ -21,40 +24,122 @@
 
 NULL_DSET_ID = b"00000000000000000000000000000000"
 
+freq = [3, 50, 777, 554]
+
+# frequency to set lost samples
+frac_freq = -2
+frac_lost = 0.8
+frac_rfi = 0.3
+
+# frequency to set empty frames
+empty_freq = 3
+
+# gains parameters
+start_time = 1_500_000_000
+
+old_timestamp = start_time - 10.0
+new_timestamp = start_time + 5.0
+
+old_update_id = f"gains{old_timestamp}"
+new_update_id = f"gains{new_timestamp}"
+
+transition_interval = 10.0
+new_state = True
+
 writer_params = {
     "num_elements": 4,
     "num_ev": 2,
-    "cadence": 5.0,
-    "total_frames": 10,  # One extra sample to ensure we actually get 256
-    "freq": [3, 50, 777, 554],
+    "cadence": 1.0,
+    "total_frames": 10,
+    "freq": freq,
     "chunk_size": [2, 6, 5],
     "mode": "test_pattern_simple",
     "test_pattern_value": [0, 0],
     "file_type": "hdf5fast",
     "dataset_manager": {"use_dataset_broker": False},
+    "updatable_config": "/gains",
+    "use_local_dataset_man": True,
+    "gains": {
+        "kotekan_update_endpoint": "json",
+        "start_time": old_timestamp,
+        "update_id": old_update_id,
+        "transition_interval": transition_interval,
+        "new_state": new_state,
+    },
 }
 
 stack_params = {
     "num_elements": 2048,
     "num_ev": 2,
     "cadence": 5.0,
     "file_length": 3,
-    "freq": [3, 50, 777, 554],
+    "freq": freq,
     "chunk_size": [2, 64, 3],
     "dataset_manager": {"use_dataset_broker": False},
 }
 
-# frequency to set lost samples
-frac_freq = -2
-frac_lost = 0.8
-frac_rfi = 0.3
+# Simulate a basic cal broker
```

Comment:
This, and a bunch of the code below looks like it's just been copied out of `test_applygains.py` can you factor it into a common fixture they both use.

Reply:
at first I wasn't sure if we wanted each test to be more or less independent, but if this is preferable I can import it from `test_applygains`